src/main/java/betterquesting/api2/storage/LookupLogicType.java

```diff
@@ -0,0 +1,37 @@
+package betterquesting.api2.storage;
+
+import java.util.function.Function;
+import java.util.function.Predicate;
+
+import static betterquesting.api2.storage.SimpleDatabase.*;
+
+enum LookupLogicType
+{
+	ArrayCache(db -> db.mapDB.size() < CACHE_MAX_SIZE && db.mapDB.size() > SPARSE_RATIO * (db.mapDB.lastKey() - db.mapDB.firstKey()), ArrayCacheLookupLogic::new),
+	SortMergeJoin(db -> db.mapDB.size() > SORT_MERGE_JOIN_THRESHOLD, SortMergeJoinLookupLogic::new),
```

Comment:
do we need SortMergeJoin now? HashMap was always better in my benchmark
but i didnt test with TIntObjectMap yet (i think should be the same or maybe a bit faster)

Reply:
Yeah both HashMap and TIntObjectHashMap works faster than sort merge join in any situation. I removed it in the latest commit.